x/sync/manager.go

```diff
@@ -169,8 +168,8 @@ type Manager struct {
 }
 
 // TODO remove non-config values out of this struct
-type ManagerConfig struct {
-	Parser                ProofParser
+type ManagerConfig[T, U any] struct {
+	ProofHandler          ProofHandler[T, U]
```

Comment:
We could partially satisfy the above TODO by making this a field/parameter into `NewManager` instead of a part of this config - which will also remove the generic signature on the config type.